src/provider/provider_file_memory.c

```diff
@@ -0,0 +1,618 @@
+    }
+
+    void *value =
+        critnib_remove(file_provider->fd_offset_map, (uintptr_t)highPtr);
+    if (value == NULL) {
+        LOG_ERR("file_allocation_merge(): removing a value from the file "
+                "descriptor offset map failed (addr=%p)",
+                highPtr);
+        return UMF_RESULT_ERROR_UNKNOWN;
+    }
+
+    return UMF_RESULT_SUCCESS;
+}
+
+typedef struct os_ipc_data_t {
+    char path[NAME_MAX];
+    size_t offset_fd;
+    size_t size;
+} os_ipc_data_t;
+
+static umf_result_t file_get_ipc_handle_size(void *provider, size_t *size) {
+    if (provider == NULL || size == NULL) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    *size = sizeof(os_ipc_data_t);
+
+    return UMF_RESULT_SUCCESS;
+}
+
+static umf_result_t file_get_ipc_handle(void *provider, const void *ptr,
+                                        size_t size, void *providerIpcData) {
+    if (provider == NULL || ptr == NULL || providerIpcData == NULL) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    file_memory_provider_t *file_provider = (file_memory_provider_t *)provider;
+
+    void *value = critnib_get(file_provider->fd_offset_map, (uintptr_t)ptr);
+    if (value == NULL) {
+        LOG_ERR("file_get_ipc_handle(): getting a value from the IPC cache "
+                "failed (addr=%p)",
+                ptr);
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    os_ipc_data_t *os_ipc_data = (os_ipc_data_t *)providerIpcData;
+    os_ipc_data->offset_fd = (size_t)value - 1;
+    os_ipc_data->size = size;
+    strncpy(os_ipc_data->path, file_provider->path, NAME_MAX - 1);
+    os_ipc_data->path[NAME_MAX - 1] = '\0';
+
+    return UMF_RESULT_SUCCESS;
+}
+
+static umf_result_t file_put_ipc_handle(void *provider, void *providerIpcData) {
+    if (provider == NULL || providerIpcData == NULL) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    file_memory_provider_t *file_provider = (file_memory_provider_t *)provider;
+    os_ipc_data_t *os_ipc_data = (os_ipc_data_t *)providerIpcData;
+
+    if (strncmp(os_ipc_data->path, file_provider->path, NAME_MAX)) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    return UMF_RESULT_SUCCESS;
+}
+
+static umf_result_t file_open_ipc_handle(void *provider, void *providerIpcData,
+                                         void **ptr) {
+    if (provider == NULL || providerIpcData == NULL || ptr == NULL) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    file_memory_provider_t *file_provider = (file_memory_provider_t *)provider;
+    os_ipc_data_t *os_ipc_data = (os_ipc_data_t *)providerIpcData;
+    umf_result_t ret = UMF_RESULT_SUCCESS;
+    int fd;
+
+    if (strncmp(os_ipc_data->path, file_provider->path, NAME_MAX)) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    fd = os_file_open(os_ipc_data->path);
+    if (fd <= 0) {
+        LOG_PERR("opening a shared memory file (%s) failed", os_ipc_data->path);
```

Comment:
hmm.. I gues we should have negative tests for IPC, like e.g. here, for improper path

Reply:
Hmm ...

Reply:
Done